libraries/AP_InertialSensor/AP_InertialSensor.cpp

```diff
@@ -1950,12 +1950,24 @@ void AP_InertialSensor::update(void)
         for (uint8_t i=0; i<INS_MAX_INSTANCES; i++) {
             if (_gyro_healthy[i] && _use(i)) {
                 _first_usable_gyro = i;
+#if AP_AHRS_ENABLED
+                // ask AHRS for the true primary, might just be us though
+                _primary_gyro = AP::ahrs().get_primary_gyro_index();
+#else
+                _primary_gyro = _first_usable_gyro;
+#endif
```

Comment:
Shouldn't this be outside the for loop?

Reply:
its inside because I need _first_usable_gyro to be definitely set. Could split this up though with a #define if you think that's better?

Reply:
Fixed